packages/backend/src/index.ts

```diff
@@ -3,15 +3,21 @@ import { getConfig } from './config'
 
 const env = process.env.NODE_ENV === 'production' ? 'production' : 'local'
 
-try {
-  const config = getConfig(env)
-  const app = new Application(config)
-  app.start().catch(onError)
-} catch (e) {
-  onError(e)
+if (require.main === module) {
+  try {
+    const config = getConfig(env)
+    const app = new Application(config)
+    app.start().catch(onError)
+  } catch (e) {
+    onError(e)
+  }
 }
 
 function onError(e: unknown) {
   console.error(e)
   process.exit(1)
 }
+
+// Public API exposed for encoding tests
+
+export { getOnChainData } from './peripherals/onchain'
```

Comment:
Maybe we should do it differently. Instead of having `src/index.ts` we could have `src/main.ts` and `src/lib.ts`. Then we'd set package.json's `main` to `lib.js`